samples/isprime/isprime.neon

```diff
@@ -0,0 +1,184 @@
+    VAR a:       Number := 0
+    VAR i:       Number := iIterations
+
+    IF NOT isOdd(n) THEN
+        % Rule: The number n is an EVEN number, and therefore CANNOT be prime.
+        IF n = 2 THEN
+            % The number 2 is the only exception to the rule!
+            % Since 2 is the ONLY even number that is prime, we handle this
+            % case a little differently.
+            RETURN TRUE
+        END IF
+        RETURN isprime
+    END IF
+
+    IF m > 0 THEN
+        WHILE binary.get32(m, 0) = TRUE DO
+            INC k
+            m := binary.shiftRight32(m, 1)
+            IF m < 0 THEN 
+                EXIT WHILE
+            END IF
+        END WHILE
+    END IF
+
+    VAR r: Number
+    REPEAT
+        GetRandomA(OUT a, n)
+        r := 1
+        FOR x := 0 TO m - 1 DO
+            r := (r * a) MOD n
+        END FOR
+        IF r = 1 OR r = (n - 1) THEN
+            isprime := TRUE
+            DEC i
+        END IF
+        FOR x := 0 TO k DO
+            r := (r*r) MOD n
+            IF r = (n - 1) THEN
+                isprime := TRUE
+                EXIT FOR
+            END IF
+        END FOR
+        IF NOT isprime THEN
+            EXIT REPEAT
+        END IF
+        DEC i
+    UNTIL i # 0
+    RETURN isprime
+END FUNCTION
+
+FUNCTION SmallFactor(n: Number): Boolean
+    FOR i := 0 TO SizeOf(SMALL_PRIME)-1 DO
+        IF n = SMALL_PRIME[i] THEN
+            EXIT FOR
+        END IF
+        IF (n MOD SMALL_PRIME[i] = 0) THEN
+            RETURN FALSE
+        END IF
+    END FOR
+    RETURN TRUE
+END FUNCTION
+
+BEGIN MAIN
+    IF sys.args.size() <= 1 THEN
+        print("\nUsage: \(sys.args[0]) {Number | s-e | s- | -e}")
+        print(" Where: s is starting integer, and e is ending integer.\n")
+        sys.exit(1)
+    END IF
+
+    VAR sval: Number := 0
+    VAR eval: Number := 0
+    VAR hasDash: Number := string.find(sys.args[1], "-")
+    IF string.hasPrefix(sys.args[1], "-") THEN
+        % Print the last prime number BEFORE n.
+        eval := 3
+        sval := num(substring(sys.args[1], hasDash+1, sys.args[1].length()-hasDash-1))
+        print("Locating last prime number before \(sval)...")
+        VAR val: Number := 0
+        FOR x := sval TO eval STEP -1 DO
+            val := x
+            IF RabinMiller(val, TestIterations) THEN
+                print("\(val)")
+                EXIT FOR
+            END IF
+            IF isOdd(val) THEN
+                val := val - 2
+            ELSE
+                DEC val
+            END IF
+        END FOR
+    ELSIF string.hasSuffix(sys.args[1], "-") THEN
+        % Print the next prime AFTER n.
+        IF hasDash > 0 THEN
+            sval := num(substring(sys.args[1], 0, hasDash))
+            print("Locating the next prime number after \(sval)...")
+            WHILE TRUE DO
+                IF RabinMiller(sval, TestIterations) THEN
+                    EXIT WHILE
+                END IF
+                IF isOdd(sval) THEN
```

Comment:
This won't work if you ask for the primes from "1-", since 1 is odd so you'll increment `sval` to 3 and skip 2 (which is prime).